specification/columns/commitmentdiscountunit.md

```diff
@@ -7,8 +7,8 @@ The CommitmentDiscountUnit column adheres to the following requirements:
 * CommitmentDiscountUnit MUST be present in a FOCUS dataset when the provider supports [*commitment discounts*](#glossary:commitment-discount).
```

Comment:
Per Chris' suggestion, adding this as a suggestion rather than just a comment. Since I can only provide suggestions on the code snippets Chris originally changed in this PR, this is the best I can do for now. :)

---

Compared to the original, in this version:

* All the requirements presented as bullet points and sub-bullets
* MUST NOT / MAY / MUST be null across various ChargeCategory, commitmentDiscountId, ChargeClass values precisely specified.
* "MUST be the same across all *rows*" requirement moved to the "CommitmentDiscountUnit is not null" block

---


```suggestion
* CommitmentDiscountUnit MUST be present in a FOCUS dataset when the provider supports [*commitment discounts*](#glossary:commitment-discount).
* CommitmentDiscountUnit MUST be of type String.
* Units of measure used in CommitmentDiscountUnit SHOULD adhere to the values and format requirements specified in the [UnitFormat](#unitformat) attribute.
* CommitmentDiscountUnit MUST NOT be null when [ChargeCategory](#chargecategory) is "Purchase" or "Usage", [CommitmentDiscountId](#commitmentdiscountid) is not null, and [*ChargeClass*](#chargeclass) is not "Correction".
* CommitmentDiscountUnit MAY be null when ChargeCategory is "Purchase" or "Usage", CommitmentDiscountId is not null, and ChargeClass is "Correction".
* CommitmentDiscountUnit MUST be null in all other cases.
* In cases where the CommitmentDiscountUnit is not null, the following applies:
  * CommitmentDiscountUnit MUST be the same across all *rows* having the same [CommitmentDiscountId](#commitmentdiscountid).
  * CommitmentDiscountUnit MUST represent the unit used to measure the *commitment discount*.
  * When accounting for [*commitment discount flexibility*](glossary:commitment-discount-flexibility), the CommitmentDiscountUnit value SHOULD reflect this consideration.
```

Reply:
Discussed in TF-1 today, going to revisit a more wide-ranging readability topic in a later release